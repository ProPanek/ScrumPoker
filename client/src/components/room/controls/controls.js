import React, {Component} from "react"
import {inject, observer} from "mobx-react";
import styled from "styled-components";
import Button from "@material-ui/core/Button";
import ButtonDnd from "./controlsbutton_dnd"

const ButtonWrapper = styled.div`
  display: block;
  margin: 0 auto;
  width: 100%;
  height: 10vh;
`;

const cards = ["â˜•", 0, 0.5, 1, 2, 3, 5, 8, 13, 20, 40, 100]

class Controls extends Component {
  constructor(props) {
    super(props)
    this.selectedCard = ""
  }

  handleReset = () => {
    this.props.store.roomStore.resetCards()
  }

  handleCard = () => {
    const {store: {roomStore: {sendCard, setBlockCard}}} = this.props
    sendCard()
    setBlockCard(true)
  }

  render() {
    const {store: {roomStore: {blockCard, blockReset, cardsAreTheSame}, userStore: {admin}, jiraStore: {jiraLoggedIn}}} = this.props
    return (
      <>
        <Button
          disabled={blockCard}
          variant="contained"
          color="secondary"
          onClick={this.handleCard}>
          Send Card
        </Button>
        {admin && (
          <>
            <Button
              disabled={blockReset}
              onClick={this.handleReset}>Reset cards</Button>
            {jiraLoggedIn && <Button
              disabled={!cardsAreTheSame}
              onClick={this.handleEstimation}>Set estimation point</Button>
            }
          </>
        )}
        <ButtonWrapper>
          {cards.map((card, index) => {
            return (
                <ButtonDnd key={index} value={card}/>
            )
          })}
        </ButtonWrapper>
      </>
    )
  }
}

export {Controls}
export default inject("store")(observer(Controls));

