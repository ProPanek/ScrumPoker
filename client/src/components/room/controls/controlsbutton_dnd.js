import React, {Component} from 'react'
import {getEmptyImage} from "react-dnd-html5-backend";

import {
  DragSource,
} from 'react-dnd'

const Types = {
  ITEM: "cardButton"
}

const buttonSource = {
  beginDrag(props) {
    return {
      value: props.value
    };
  },
  endDrag(props, monitor, component) {
    console.log(monitor.didDrop())
  }
};

function collect(connect, monitor) {
  return {
    connectDragSource: connect.dragSource(),
    isDragging: monitor.isDragging(),
    didDrop: monitor.didDrop(),
    connectDragPreview: connect.dragPreview(),
  }
}

class ButtonDnd extends Component {

  componentDidMount() {
    const {connectDragPreview} = this.props;
    connectDragPreview(getEmptyImage());
  }

  render() {
    const {connectDragSource, value, isDragging} = this.props;

    const style = {
      // display: display ? "initial" : "none",
      backgroundColor: isDragging ? "white" : "#3f51b5",
      border: isDragging ? "1px dashed black" : "none",
      height: "10vh",
      margin: "1px",
      boxShadow: "0px 1px 5px 0px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 3px 1px -2px rgba(0,0,0,0.12)",
      flex: 1,
      padding: "6px 16px",
      fontSize: "0.875rem",
      minWidth: "64px",
      boxSizing: "border-box",
      transition: "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
      lineHeight: "1.75",
      fontWeight: "500",
      fontFamily: "'Roboto', 'Helvetica', 'Arial', sans-serif",
      borderRadius: "4px",
      textTransform: "uppercase",
      color: "#fff",
      alignItems: "center",
      verticalAlign: "middle",
      justifyContent: "center",
      outline: "0"
    }

    return connectDragSource(
      <button
        style={style}>{isDragging ? "" : value}</button>
    );
  }
}

export default DragSource(Types.ITEM, buttonSource, collect)(ButtonDnd);
