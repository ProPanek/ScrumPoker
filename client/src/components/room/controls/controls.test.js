import React from "react"
import { Controls } from "./controls"
import { shallow } from "enzyme";

describe("Controls", () => {
  it("renders without errors", () => {
    const store = {
      roomStore: { cardsAreTheSame : false, cardResults : []},
      userStore: {admin: true},
      jiraStore: {jiraLoggedIn: true}
    }

    const component = shallow(<Controls store={store} />)
    expect(component).toMatchSnapshot()
  })
})
