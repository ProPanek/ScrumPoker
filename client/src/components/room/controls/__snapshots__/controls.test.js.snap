// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls renders without errors 1`] = `
<Fragment>
  <WithStyles(Button)
    color="secondary"
    onClick={[Function]}
    variant="contained"
  >
    Send Card
  </WithStyles(Button)>
  <WithStyles(Button)
    onClick={[Function]}
  >
    Reset cards
  </WithStyles(Button)>
  <WithStyles(Button)
    disabled={true}
  >
    Set estimation point
  </WithStyles(Button)>
  <styled.div>
    <DragSource(ButtonDnd)
      key="0"
      value="â˜•"
    />
    <DragSource(ButtonDnd)
      key="1"
      value={0}
    />
    <DragSource(ButtonDnd)
      key="2"
      value={0.5}
    />
    <DragSource(ButtonDnd)
      key="3"
      value={1}
    />
    <DragSource(ButtonDnd)
      key="4"
      value={2}
    />
    <DragSource(ButtonDnd)
      key="5"
      value={3}
    />
    <DragSource(ButtonDnd)
      key="6"
      value={5}
    />
    <DragSource(ButtonDnd)
      key="7"
      value={8}
    />
    <DragSource(ButtonDnd)
      key="8"
      value={13}
    />
    <DragSource(ButtonDnd)
      key="9"
      value={20}
    />
    <DragSource(ButtonDnd)
      key="10"
      value={40}
    />
    <DragSource(ButtonDnd)
      key="11"
      value={100}
    />
  </styled.div>
</Fragment>
`;
